packages/components/addon/components/hds/app-footer/status-link.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+export const STATUSES = {
+  OPERATIONAL: {
+    text: 'Operational',
+    iconName: 'check-circle-fill',
+    iconColor: 'var(--token-color-foreground-success)',
+  },
+  DEGRADED: {
+    text: 'Degraded',
+    iconName: 'alert-triangle',
+    iconColor: 'var(--token-color-foreground-warning)',
+  },
+  OUTAGE: {
+    text: 'Outage',
+    iconName: 'x-square-fill',
+    iconColor: 'var(--token-color-foreground-critical)',
+  },
+  MAINTENANCE: {
+    text: 'Maintenance',
+    iconName: 'info-fill',
+    iconColor: 'var(--token-color-foreground-highlight)',
+  },
+};
+
+export default class HdsAppFooterStatusLinkComponent extends Component {
+  /**
+   * @param ariaLabel
+   * @type {string}
+   * @description The name of the status which the StatusLink is being set to
+   */
+  get status() {
+    return this.args.status.toUpperCase();
```

Comment:
[suggestion] I wonder if we should do something similar to how we handle options for the button component, for example. If they don't pass one of the allowed options we throw an error?

Reply:
Thanks, that's what I'm working on doing now. Will push up after I finish a few last things.

Reply:
Just pushed up a bunch of refactors around these. We may still want to make other changes though. I'm not sure of all the possible use cases and what combo of args should be required vs. not.

Reply:
Going to mark this as resolved, thanks.